lib/base/networking/api/handler/stream_handler.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:fixnum/fixnum.dart';
+import 'package:gocast_mobile/base/networking/api/gocast/api_v2.pb.dart';
+import 'package:gocast_mobile/base/networking/api/handler/grpc_handler.dart';
+import 'package:logger/logger.dart';
+
+class StreamHandler {
+  static final Logger _logger = Logger();
+  final GrpcHandler _grpcHandler;
+
+  StreamHandler(this._grpcHandler);
+
+  /// Fetches the current course streams.
+  ///
+  /// This method sends a `getUserStreams` gRPC call to fetch the user's
+  /// streams.
+  ///
+  /// Takes [courseID] as a parameter.
+  /// Returns a [List<Stream>] instance that represents the user's streams.
+  Future<List<Stream>> fetchCourseStreams(int courseID) async {
+    _logger.i('Fetching streams');
+    return _grpcHandler.callGrpcMethod(
+      (client) async {
+        final response = await client
+            .getCourseStreams(GetCourseStreamsRequest(courseID: courseID));
+        _logger.d('Streams: ${response.streams}');
+        return response.streams;
+      },
+    );
+  }
+
+  /// Fetches the stream of the course.
+  ///
+  /// This method sends a `getStream` gRPC call to fetch the stream of the course.
+  ///
+  /// Takes [streamId] as a parameter.
+  ///
+  /// Returns a [Stream] instance that represents the stream of the course.
+  Future<Stream> fetchStream(Int64? streamId) async {
```

Comment:
Is there a reason why the streamID is marked as optional? ```Int64?```

Reply:
Fix ✅